crates/starknet-rs-py/src/starknet_state.rs

```diff
@@ -1 +1,181 @@
+        let signature = signature.map(|signs| {
+            signs
+                .into_iter()
+                .map(Felt252::from)
+                .collect::<Vec<Felt252>>()
+        });
+
+        let nonce = nonce.map(Felt252::from);
+
+        let exec_info = self
+            .inner
+            .invoke_raw(address, selector, calldata, max_fee, signature, nonce)
+            .map_err(|e| PyRuntimeError::new_err(e.to_string()))?;
+
+        Ok(PyTransactionExecutionInfo::from(exec_info))
+    }
+
+    pub fn execute_entry_point_raw(
+        &mut self,
+        contract_address: BigUint,
+        entry_point_selector: BigUint,
+        calldata: Vec<BigUint>,
+        caller_address: BigUint,
+    ) -> PyResult<PyCallInfo> {
+        let calldata = calldata
+            .into_iter()
+            .map(Felt252::from)
+            .collect::<Vec<Felt252>>();
+        let entry_point_selector = Felt252::from(entry_point_selector);
+
+        let call_info = self
+            .inner
+            .execute_entry_point_raw(
+                Address(Felt252::from(contract_address)),
+                entry_point_selector,
+                calldata,
+                Address(Felt252::from(caller_address)),
+            )
+            .map_err(|e| PyRuntimeError::new_err(e.to_string()))?;
+
+        Ok(PyCallInfo::from(call_info))
+    }
+
+    pub fn deploy(
+        &mut self,
+        contract_class: &PyContractClass,
+        constructor_calldata: Vec<BigUint>,
+        contract_address_salt: BigUint,
+    ) -> PyResult<(BigUint, PyTransactionExecutionInfo)> {
+        let contract_class = contract_class.inner.clone();
+        let constructor_calldata = constructor_calldata
+            .into_iter()
+            .map(Felt252::from)
+            .collect::<Vec<Felt252>>();
+        let contract_address_salt = Address(Felt252::from(contract_address_salt));
+
+        let (address, exec_info) = self
+            .inner
+            .deploy(contract_class, constructor_calldata, contract_address_salt)
+            .map_err(|e| PyRuntimeError::new_err(e.to_string()))?;
+        Ok((
+            address.0.to_biguint(),
+            PyTransactionExecutionInfo::from(exec_info),
+        ))
+    }
+
+    pub fn execute_tx(&mut self, tx: &mut PyTransaction) -> PyResult<PyTransactionExecutionInfo> {
+        let tx_info = self
+            .inner
+            .execute_tx(&mut tx.inner)
+            .map_err(|e| PyRuntimeError::new_err(e.to_string()))?;
+        Ok(PyTransactionExecutionInfo::from(tx_info))
+    }
+
+    pub fn consume_message_hash(&mut self, message_hash: Vec<u8>) -> PyResult<()> {
+        self.inner
+            .consume_message_hash(message_hash)
+            .map_err(|e| PyRuntimeError::new_err(e.to_string()))?;
+        Ok(())
+    }
+
+    fn set_contract_class(
+        &mut self,
+        address: BigUint,
+        contract_class: &PyContractClass,
+    ) -> PyResult<()> {
+        let hash = felt_to_hash(&Felt252::from(address));
```

Comment:
`address` would be the class hash right? Also, we can use `ClassHash` instead of `BigUint`, as it's just an alias for an array.

Reply:
address is a biguint for python, in this case is the same to use like that or as an array. I would leave it as a biguint because as far as i have seen, is more common to handle addresses like numbers than like arrays.

Reply:
i must have mixed myself with other call. fixed.